README.md

```diff
@@ -1 +1,976 @@
+configurations we will create:
+```
+$ rm ~/.balenarc.yml
+$ rm -rf ~/.balena
+```
+
+Now, in your home directory, create two directories called `balenaProduction`
+and `balenaStaging`, and then fill in separate configuration files in each
+directory. The following commands will do this for you:
+```
+$ mkdir -p ~/balenaProduction
+$ mkdir -p ~/balenaStaging
+$ echo "balenaUrl: 'balena-staging.com'" > ~/balenaStaging/balenarc.yml
+$ echo "balenaUrl: 'balena-cloud.com'" > ~/balenaProduction/balenarc.yml
+```
+Note that we do *not* prefix the `balenarc.yml` file with a `.` to make it
+hidden.
+
+We can now use the `BALENARC_DATA_DIRECTORY` to tell balenaCLI which directory
+to store data such as the authentication token for the current environment.
+Using this and then logging into the relevant service will store an
+authentication token in the appropriate directory. Run the following in a
+terminal to login to the production environment:
+```
+$ cd ~/balenaProduction
+$ BALENARC_DATA_DIRECTORY=. balena login
+ _            _
+| |__   __ _ | |  ____  _ __    __ _
+| '_ \ / _` || | / __ \| '_ \  / _` |
+| |_) | (_) || ||  ___/| | | || (_) |
+|_.__/ \__,_||_| \____/|_| |_| \__,_|
+
+
+Logging in to balena-cloud.com
+? How would you like to login? (Use arrow keys)
+❯ Web authorization (recommended)
+  Credentials
+  Authentication token
+  I don't have a balena account!
+```
+Login using your preferred method. Note that a new `token` file now exists in
+the `~/balenaProduction` directory. Now login to the staging directory, using
+the staging configuration directory:
+```
+$ cd ~/balenaStaging
+$ BALENARC_DATA_DIRECTORY=. balena login
+ _            _
+| |__   __ _ | |  ____  _ __    __ _
+| '_ \ / _` || | / __ \| '_ \  / _` |
+| |_) | (_) || ||  ___/| | | || (_) |
+|_.__/ \__,_||_| \____/|_| |_| \__,_|
+
+
+Logging in to balena-staging.com
+? How would you like to login? (Use arrow keys)
+❯ Web authorization (recommended)
+  Credentials
+  Authentication token
+  I don't have a balena account!
+```
+This time, a `token` file was created in the `~/balenaStaging` directory.
+
+You can now switch between environments by changing directory to the one
+with the relevant `balenarc.yml` and `token` file, which will allow you
+to use either environment without any further authentication:
+```
+$ cd ~/balenaStaging
+$ BALENARC_DATA_DIRECTORY=. balena apps
+ID    APP NAME          DEVICE TYPE  ONLINE DEVICES DEVICE COUNT
+12345 appOne            intel-nuc    0              0
+12346 appTwo            iot2000      0              0
+$ cd ~/balenaProduction
+$ BALENARC_DATA_DIRECTORY=. balena apps
+ID      APP NAME         DEVICE TYPE      ONLINE DEVICES DEVICE COUNT
+54321   artik530         artik533s        0              0
+54322   bob              intel-nuc        0              0
+54323   bbbtest          beaglebone-black 0              0
+```
+
+We'll be using the two separate environments in the next set of exercises
```

Comment:
Not sure they're good for public consumption really ;)

Reply:
Yeah, I'd rather stick to actual commands here rather than 'bash magic'. If users want to do scripting that's fine, but we really shouldn't be adding scripts here.

Reply:
Would be a worthwhile hack friday to add multiple environment support to `balena-cli` mind. I envisage something similar to how aws cli tooling works.

Reply:
@wrboyce Dude, that would be utterly awesome